File: Car-Rental-BE/src/main/java/com/gi2/carrental/entity/Order.java
package com.gi2.carrental.entity;

import com.gi2.carrental.Enums.Statu;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDate;
@Data
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "orders")
public class Order {
    @Id
    private String id;
    @DBRef
    private Client client;
    @DBRef
    private Car car;
    private Statu State;
    private LocalDate startDate;
    private LocalDate endDate;
    private double totalCost;
}

File: Car-Rental-FE/src/app/entities/order.ts
import { Car } from "./car";
import { Client } from "./client";

export interface Order {
    id: any;
    client: Client;
    car: Car;
    state: Statu;
    startDate: string; // You can use the 'string' type or a custom type for dates
    endDate: string; // You can use the 'string' type or a custom type for dates
    totalCost: number;
  }

  export enum Statu {
    Accepte = 'Accepte',
    Refuse = 'Refuse',
    InHold = 'InHold'
  }